### PR TITLE
Add goerr113 linter check

### PR DESCRIPTION
**Pull request by hj-johannes-lee**
Add goerr113 lintercheck
Fix the usage of fmt.Errorf() by wrapping errors
Fix the usage of errors.New()

Closes: #641

### CODE REVIEW

**Review by rojkov — COMMENTED** [review]
Thanks! This is good. Just get rid of the new error types which are not used for error handling in the rest of the code.

File: test/e2e/utils/utils.go [hunk 1]
```diff
@@ -79,14 +79,24 @@ func WaitForPodFailure(f *framework.Framework, name string, timeout time.Duratio
 			case v1.PodFailed:
 				return true, nil
 			case v1.PodSucceeded:
-				return true, fmt.Errorf("pod %q successed with reason: %q, message: %q", name, pod.Status.Reason, pod.Status.Message)
+				return true, errPodSucceeded{name, pod.Status.Reason, pod.Status.Message}
```
Comment (rojkov) [inline test/e2e/utils/utils.go:82]: It's enough to use `errors.Errorf()` instead of `fmt.Errorf()` here.

**Review by rojkov — COMMENTED** [review]
Looks great now!

Could you please also enable goerr113 in `.golangci.yaml`?

File: cmd/fpga_crihook/main.go [hunk 2]
```diff
@@ -207,25 +207,51 @@ func getStdin(reader io.Reader) (*Stdin, error) {
...
+		return nil, errAnnotationValueNotCorrect{annotationName, stdinJ.Annotations.ComIntelFpgaMode}
 	}
 
 	if stdinJ.Bundle == "" {
 		return nil, errors.New("'bundle' field is not set in the stdin JSON")
 	}
 
 	if _, err := os.Stat(stdinJ.Bundle); err != nil {
-		return nil, fmt.Errorf("bundle directory %s: stat error: %+v", stdinJ.Bundle, err)
+		return nil, errStat{stdinJ.Bundle, err}
```
[the remark this answers is not in the document]
Comment (hj-johannes-lee) [reply cmd/fpga_crihook/main.go]: I think you menat "errors.Wrapf," right? I changed following your suggestion.!

[the remark this answers is not in the document]
Comment (hj-johannes-lee) [reply cmd/fpga_crihook/main.go]: Wrap seems to be not working. Wrapf works.!
I think I need more study about the differences of all of them.!

[the remark this answers is not in the document]
Comment (rojkov) [reply cmd/fpga_crihook/main.go]: Sorry. `Wrapf()` of course.

File: test/e2e/utils/utils.go (hunk 1 above)
Comment (hj-johannes-lee) [reply test/e2e/utils/utils.go:82]: Oops. Idk why goerr113 I put was missing in the commit.! Let me put it.!

**Review by rojkov — APPROVED** [review]
Very good!

Please make this draft be a real pull request.

**Review by mythi — APPROVED** [review]

**Pull request merged**